xbmc/addons/Addon.h

```diff
@@ -35,7 +35,7 @@ class CAddon : public IAddon
 
   TYPE Type() const override { return m_addonInfo->MainType(); }
   TYPE FullType() const override { return Type(); }
-  bool IsType(TYPE type) const override { return type == m_addonInfo->MainType(); }
+  bool IsType(TYPE type) const override { return m_addonInfo->IsType(type); }
```

Comment:
Wouldn't it be easier to:

- Change `Type()` function to `MainType()` as the function name is misleading
- Create a new method `bool HasType(Type)` that would return `true` if an addon has the given type
- Just use HasType(xx) in the switch/if conditions

?

Reply:
With `Type()`I can change, was only leaved as them to have the cpluff remove request not bigger with rename of them.

With new method `bool HasType(Type)`, what should be different to `bool IsType(TYPE type)`? 

Or you mean to leave the ` { return type == m_addonInfo->MainType(); }`, this was I think only added from me to prevent this view the 2 new types on addon Window (If I remember correctly)?

Reply:
I just gave it a try and it does indeed fix the issue with `screensaver.atv4`. However I don't like the Scripts and Media sources showing in the addon browser.

What I mean with my suggestion was exactly that. `Type()` and `IsType()` refer to MainType only so they should be renamed to reflect what they do. `HasType` (or your new `IsType`) would have the implementation you provide in this PR. The only code block using `HasType` would be `GetScriptsAndPlugins` for now.